src/hooks/useFormOnUpdate.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { usePrevious } from '@gridsuite/commons-ui';
+import { useEffect } from 'react';
+import isDeepEqualReact from 'fast-deep-equal/react';
+import { UseFormReturn } from 'react-hook-form';
+
+const useFormOnUpdate = (formApi: UseFormReturn, data: any) => {
+    const prevData = usePrevious(data);
+
+    useEffect(() => {
+        if (!isDeepEqualReact(prevData, data)) {
+            const setValueToKey = ([key, value]: [string, any]) => {
```

Comment:
[eslint] 
 React Hook useEffect was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies

 React Hook useEffect has missing dependencies: 'data' and 'formApi'. Either include them or remove the dependency array

Reply:
I use ignore eslint : // eslint-disable-next-line react-hooks/exhaustive-deps, this is a particular case in which we force reset when mapping or equipment type switched